x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/expression/function/scalar/datetime/DateTimeToCharProcessorTests.java

```diff
@@ -22,43 +27,56 @@
+ *     <li>Spin up a Postgres instance (latest or a specific version) using docker:
+ *       <pre>
+ *       docker run --rm --name postgres-latest -e POSTGRES_PASSWORD=mysecretpassword -p 5432:5432 -d postgres:latest
+ *       </pre>
+ *     </li>
+ *     <li>Generate the test dataset by execution the SQL against PostgreSQL and capturing the output:
+ *       <pre>
+ *       PGPASSWORD="mysecretpassword" psql --quiet -h localhost -p 5432 -U postgres -f /tmp/postgresql-tochar-test.sql \
+ *           &gt; /path/to/tochar-generated.csv
+ *       </pre>
+ *     </li>
+ * </ol>
+ * 
+ * In case you need to mute any of the tests, mute all tests by adding {@link org.apache.lucene.util.LuceneTestCase.AwaitsFix}
+ * on the class level.
+ */
 public class DateTimeToCharProcessorTests extends ESTestCase {
-
-    /**
-     * Tests the {@link ToCharFormatter} against actual PostgreSQL output.
-     * 
-     * Process to (re)generate the test data:
-     * <ol>
-     *     <li>Run the @{link {@link ToCharTestScript#main(String[])}} class</li>
-     *     <li>Spin up a Postgres instance (latest or a specific version) using docker:
-     *       <pre>
-     *       docker run --rm --name postgres-latest -e POSTGRES_PASSWORD=mysecretpassword -p 5432:5432 -d postgres:latest
-     *       </pre>
-     *     </li>
-     *     <li>Generate the test dataset by execution the SQL against PostgreSQL and capturing the output:
-     *       <pre>
-     *       PGPASSWORD="mysecretpassword" psql --quiet -h localhost -p 5432 -U postgres -f /tmp/postgresql-tochar-test.sql \
-     *           &gt; /path/to/tochar-generated.csv
-     *       </pre>
-     *     </li>
-     * </ol>
-     */
-    public void testAgainstPostgreSQLOutput() {
+    
+    @ParametersFactory(argumentFormatting = "%1$s:%2$s %5$s")
+    public static Iterable<Object[]> parameters() throws Exception {
+        List<Object[]> params = new ArrayList<>();
         String testFile = "tochar-generated.csv";
         int lineNumber = 0;
-        for (String line : ToCharTestScript.readAllLinesWithoutComment(DateTimeToCharProcessorTests.class.getResource(testFile))) {
+        for (String line : Files.readAllLines(Path.of(DateTimeToCharProcessorTests.class.getResource(testFile).toURI()))) {
             lineNumber += 1;
             if (line.startsWith("#")) {
                 continue;
             }
             String[] cols = line.split(quote(DELIMITER));
-            testOneCase(testFile, lineNumber, cols[0], cols[1], cols[2], cols[3], cols[4]);
+            params.add(new Object[]{testFile, lineNumber, cols[0], cols[1], cols[2], cols[3], cols[4]});
         }
+        return params;
     }
 
+    private final String testFile;
+    private final int lineNumber;
+    private final String secondsAndFractionsSinceEpoch;
+    private final String zone;
+    private final String formatString;
+    private final String posgresTimestamp;
+    private final String expectedResult;
```

Comment:
I believe it is possible to pass a rich object instead of primitives inside Object[] - not sure whether formatting works.
Anyway, it's just a note for future reference.